### ISSUE TITLE
action fails with <bantime> error BUG

### ISSUE THREAD

**Issue by srulikuk** [fail2ban/fail2ban#2869]
Fail2Ban v0.11.1 on ClearOS
Just checked my f2b logs and it hasnt been working for the past few months, i have millions of errors on my log files like 
```
2020-11-09 19:31:37,132 fail2ban.utils          [17395]: ERROR   7f03602a0928 -- stderr: "/bin/sh: -c: line 0: syntax error near unexpected token `newline'"
2020-11-09 19:31:37,132 fail2ban.utils          [17395]: ERROR   7f03602a0928 -- stderr: "/bin/sh: -c: line 0: `ipset create f2b-portprobe hash:ip --maxelem 1000000 timeout <bantime>'"
2020-11-09 19:31:37,132 fail2ban.utils          [17395]: ERROR   7f03602a0928 -- returned 1
2020-11-09 19:31:37,133 fail2ban.actions        [17395]: ERROR   Failed to execute ban jail 'portprobe' action 'ipset-portprobe' info 'ActionInfo({'ip': '46.101.124.192', 'fid': <function <lambda> at 0x7f0361c71f50>, 'family': 'inet4', 'raw-ticket': <function <lambda> at 0x7f0361c74578>})': Error starting action Jail('portprobe')/ipset-portprobe: 'Script error'
```

my action file contains 
```
actionstart = ipset create <ipmset> hash:ip --maxelem 1000000 timeout <bantime><familyopt>
``` 
Replacing `<bantime>` with the actual bantime in seconds seems to work.

the action is specified in my jail as follows 
```
action = ipset-portprobe[name=portprobe,bantime=2147483]
```
It's possible that the bug is specific to my distro ClearOS, I posted the issue their too https://www.clearos.com/clearfoundation/social/community/fail2ban-bug-since-last-update

**Comment by sebres** [comment]
> my action file contains
> `actionstart = ipset create <ipmset> hash:ip --maxelem 1000000 timeout <bantime><familyopt>`

Well, it is not correct - `<bantime>` is dynamic tag in 0.11, so a ban-ticket related info (as many other tags, like `<ip>`, available in `actionban`, `actionunban`, `actionprolong` only).
At phase where command `actionstart` is executed, fail2ban doesn't know how long the ban duration is.
Although a default jail bantime would be possible, but it would make the interpolation of actions parameters more complex.
I must think about that.

Also note that internal handling of ipset timeout already caused several confusions and weird behavior (together with bantime.increment), so it was rewritten in #2703 without timeout (and ultimately fixed in 73a8175bb00ed2f456448f6edcfbf78ab7f0630e).
This will be released with next 0.10/0.11 versions.

> Replacing `<bantime>` with the actual bantime in seconds seems to work.

Sure, it remains not-substituted in command `actionstart`, so fail2ban executes `ipset create f2b-portprobe ... <bantime>` as it is, and so isn't correct at all, thus an action is not started properly (and fail2ban trying to start it each time new IP gets banned).

> It's possible that the bug is specific to my distro ClearOS

No, it is not distribution related thing. 
Just adjust your custom action like it is made in [action.d/iptables-ipset-proto6.conf](https://github.com/fail2ban/fail2ban/blob/0.11/config/action.d/iptables-ipset-proto6.conf) - so let fail2ban handle the banning/unbanning completely, without to involve ipset timeout handling.

**Comment by srulikuk** [comment]
So its not a bug but the behaviour was changed, seems to explain why it works in the older version.
I use the bantime in action so i can easily find the bantime remaining on an IP (`ipset list | grep 'x.x.x.x'`) gives me the time left, I will hard code these from now.
Many thanks sebres, resolved

**Comment by sebres** [comment]
OK, this closed.

**Comment by sebres** [comment]
Since it can be indeed considered as a certain regression, fixed now in ca4da9d1d3f792562500182711abc149b2c741a1:
in not ticket-based commands (like `actionstart` and `actionstop`) tag `<bantime>` return initial value (bantime specified in jail).
Also it is more safe now, because many other ticket-related tags will be interpolated using empty (dummy) ticket, so mostly with an empty value.
